### PR TITLE
refactor reportErrorToServer

### PR DESCRIPTION
**Pull request by tiendt**
change reportErrorToServer(message, filename, line, col, error) to reportErrorToServer(error, filename, line, col)

### CODE REVIEW

File: src/error.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ export function reportError(error, opt_associatedElement) {
 
     // 'call' to make linter happy. And .call to make compiler happy
     // that expects some @this.
-    reportErrorToServer['call'](undefined, undefined, undefined, undefined,
+    reportErrorToServer['call'](undefined, undefined, undefined,
```
Comment (lannka) [inline src/error.js]: should error be the 1st param?

File: src/error.js [hunk 2]
```diff
@@ -226,14 +226,13 @@ export function installErrorReporting(win) {
...
- * @param {string|undefined} message
+ * @param {*|undefined} error
  * @param {string|undefined} filename
  * @param {string|undefined} line
  * @param {string|undefined} col
- * @param {*|undefined} error
  * @this {!Window|undefined}
  */
-function reportErrorToServer(message, filename, line, col, error) {
+function reportErrorToServer(error, filename, line, col) {
```
Comment (lannka) [inline src/error.js]: did you miss the change in `installErrorReporting`

File: src/error.js [hunk 3]
```diff
@@ -253,7 +252,7 @@ function reportErrorToServer(message, filename, line, col, error) {
     // due to buggy browser extensions may be helpful to notify authors.
     return;
   }
-  const url = getErrorReportUrl(message, filename, line, col, error,
+  const url = getErrorReportUrl(error.message, filename, line, col, error,
```
Comment (lannka) [inline src/error.js]: we can also simplify this method now

File: src/error.js (hunk 2 above)
Comment (tiendt) [inline src/error.js]: @in installErrorReporting the function call has no parameter. All parameters are optional

Comment (tiendt) [inline src/error.js]: @lannka in installErrorReporting the function call has no parameter. All parameters are optional

**Comment by tiendt** [comment]
Closing because the current code is following window.onerror standard: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror

**Pull request closed without merging**